include/onnxruntime/core/framework/tensor.h

```diff
@@ -190,6 +190,22 @@ class Tensor final {
     shape_ = new_shape;
   }
 
+  /**
+   * Get the byte offset with respect to the p_data
+   * @warning this is a temporary solution for reusing the buffer bigger than needed.
+   */
+  inline ptrdiff_t ByteOffset() const {
+    return byte_offset_;
+  }
+
+  /**
+   * Set the byte offset with respect to the p_data
+   * @warning this is a temporary solution for reusing the buffer bigger than needed.
+   */
+  inline void SetByteOffset(ptrdiff_t byte_offset) {
+    byte_offset_ = byte_offset;
```

Comment:
If you have an instance of the Tensor class where you're going to be adjusting the byte offset, the ctor should be given some info (e.g. the full shape or an overall buffer size value) so that there can be some bounds checking when SetByteOffset is called. 

Is it the case that the byte offset would move by a fixed amount each time? e.g. say I had a batch with shape {3, 2, 2} and I had a buffer with all the data, but wanted a single Tensor instance to point to a {2, 2} section. I'd expect the valid offsets to be 0, 4 * bytes per value and 8 * bytes value. Is that the intended usage or it's more random than that?

Reply:
todo -  add a boundary check

---
In reply to: [396947051](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r396947051) [](ancestors = 396947051)

Reply:
per discussion - for current view op, which is considered to be deprecated in future, it's hard to enforce boundary check for the method or output tensor during its creation. Therefore, we created unit tests and also guard it before calling the method (as we've done in the view.cc). For those who will be using this method, for now, should do the similar check + UTs.    

---
In reply to: [396947051](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r396947051) [](ancestors = 396947051)